metrics/stackdriver_exporter.go

```diff
@@ -265,6 +279,12 @@ func sdCustomMetricsRecorder(mc metricsConfig) func(context.Context, []stats.Mea
 				return err
 			}
 			ros = append(ros, opt)
+		} else {
+			opt, err := optionForResource(metricskey.GetResource(ctx))
+			if err != nil {
+				return err
+			}
+			ros = append(ros, opt)
```

Comment:
What about:

```golang
resource := metricskey.GetResource(ctx)
if templ != nil {
  ... stuff, including creating a new resource and eventually assigning it to baseResource
}
if resource != nil {
  opt, err != optionForResource(resource)
  if err != nil {
    return err
  }
  ros = append(ros, opt)
}
```

This lets you remove the duplicate code on 277-281, and avoids adding a no-op option when `metricskey.GetResource(ctx)` returns `nil`.

Reply:
Done.